cmake/external/onnxruntime_external_deps.cmake

```diff
@@ -0,0 +1,439 @@
+)
+
+#TODO: include clog first
+if (onnxruntime_ENABLE_CPUINFO)
+  # Adding pytorch CPU info library
+  # TODO!! need a better way to find out the supported architectures
+  list(LENGTH CMAKE_OSX_ARCHITECTURES CMAKE_OSX_ARCHITECTURES_LEN)
+  if (APPLE)
+    if (CMAKE_OSX_ARCHITECTURES_LEN LESS_EQUAL 1)
+      set(CPUINFO_SUPPORTED TRUE)
+    elseif (onnxruntime_BUILD_APPLE_FRAMEWORK)
+      # We stitch multiple static libraries together when onnxruntime_BUILD_APPLE_FRAMEWORK is true,
+      # but that would not work for universal static libraries
+      message(FATAL_ERROR "universal binary is not supported for apple framework")
+    endif()
+  else()
+    # if xnnpack is enabled in a wasm build it needs clog from cpuinfo, but we won't internally use cpuinfo
+    # so we don't set CPUINFO_SUPPORTED in the CXX flags below.
+    if (onnxruntime_BUILD_WEBASSEMBLY AND NOT onnxruntime_USE_XNNPACK)
+      set(CPUINFO_SUPPORTED FALSE)
+    else()
+      set(CPUINFO_SUPPORTED TRUE)
+    endif()
+    if (WIN32)
+      # Exclude Windows ARM build and Windows Store
+      if (${onnxruntime_target_platform} MATCHES "^(ARM.*|arm.*)$" )
+        message(WARNING "Cpuinfo not included for compilation problems with Windows ARM.")
+        set(CPUINFO_SUPPORTED FALSE)
+      elseif (WIN32 AND NOT CMAKE_CXX_STANDARD_LIBRARIES MATCHES kernel32.lib)
+        message(WARNING "Cpuinfo not included non-Desktop builds")
+        set(CPUINFO_SUPPORTED FALSE)
+      endif()
+    elseif (NOT ${onnxruntime_target_platform} MATCHES "^(i[3-6]86|AMD64|x86(_64)?|armv[5-8].*|aarch64|arm64)$")
+      message(WARNING
+        "Target processor architecture \"${onnxruntime_target_platform}\" is not supported in cpuinfo. "
+        "cpuinfo not included."
+      )
+      set(CPUINFO_SUPPORTED FALSE)
+    endif()
+  endif()
+else()
+  set(CPUINFO_SUPPORTED FALSE)
+endif()
+
+if (CPUINFO_SUPPORTED)
+  if (CMAKE_SYSTEM_NAME STREQUAL "iOS")
+    set(IOS ON CACHE INTERNAL "")
+    set(IOS_ARCH "${CMAKE_OSX_ARCHITECTURES}" CACHE INTERNAL "")
+  endif()
+
+  # if this is a wasm build with xnnpack (only type of wasm build where cpuinfo is involved)
+  # we do not use cpuinfo in ORT code, so don't define CPUINFO_SUPPORTED.
+  if (NOT onnxruntime_BUILD_WEBASSEMBLY)
+    string(APPEND CMAKE_CXX_FLAGS " -DCPUINFO_SUPPORTED")
+  endif()
+
+
+  set(CPUINFO_BUILD_TOOLS OFF CACHE INTERNAL "")
+  set(CPUINFO_BUILD_UNIT_TESTS OFF CACHE INTERNAL "")
+  set(CPUINFO_BUILD_MOCK_TESTS OFF CACHE INTERNAL "")
+  set(CPUINFO_BUILD_BENCHMARKS OFF CACHE INTERNAL "")
+
+  FetchContent_Declare(
+    pytorch_cpuinfo
+    URL ${DEP_URL_pytorch_cpuinfo}
+    URL_HASH SHA1=${DEP_SHA1_pytorch_cpuinfo}
+    FIND_PACKAGE_ARGS NAMES cpuinfo
+  )
+
+endif()
+
+
+if (onnxruntime_BUILD_BENCHMARKS)
+  onnxruntime_fetchcontent_makeavailable(google_benchmark)
+endif()
+
+if (NOT WIN32)
+  #nsync tests failed on Mac Build
+  set(NSYNC_ENABLE_TESTS OFF CACHE BOOL "Build protobuf tests" FORCE)
```

Comment:
the description seems off: "Build protobuf tests"

Reply:
Cleaned it.